homeassistant/components/sonos/__init__.py

```diff
@@ -31,6 +54,21 @@
 )
 
 
+class SonosData:
+    """Storage class for platform global data."""
+
+    def __init__(self):
+        """Initialize the data."""
+        self.discovered = {}
+        self.speaker_info = {}
+        self.battery_entities = {}
+        self.media_player_entities = {}
```

Comment:
We should use the dispatching instead of storing the entities in a shared container